src/foam/core/Property.js

```diff
@@ -205,6 +205,36 @@ foam.CLASS({
           return comparePropertyValues(f(o1), f(o2));
         };
       }
+    },
+    {
+      name: 'diffPropertyValues',
+      transient: true,
+      factory: function() {
```

Comment:
Rather than build a new diff closure for every property, just use a default value and pass the property as the last parameter.

Same pattern we use for adapt/preSet/postSet etc.

Also we could refine all this in from a diff.js package